routing_model/src/scenario_tests.rs

```diff
@@ -825,6 +825,110 @@ mod dst_tests {
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_online_again() {
+        run_test(
+            "Get local event node detected online again",
+            &intial_state_old_elders(),
+            &[LocalEvent::NodeDetectedBackOnline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_votes: vec![ParsecVote::BackOnline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_offline_then_parsec_offline() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event()],
+        );
+        run_test(
+            "Get local event node detected offline and then parsec consensus offline",
+            &initial_state,
+            &[ParsecVote::Offline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_nodes: vec![NodeChange::Offline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_offline_and_parsec_offline_then_check_elder() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[
+                LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event(),
+                ParsecVote::Offline(NODE_ELDER_130).to_event(),
+            ],
+        );
+        run_test(
+            "Get parsec consensus offline then check elder",
+            &initial_state,
+            &[ParsecVote::CheckElder.to_event()],
+            &AssertState {
+                action_our_events: vec![LocalEvent::TimeoutCheckElder],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_parsec_offline_then_local_event_back_online() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[
+                LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event(),
+                ParsecVote::Offline(NODE_ELDER_130).to_event(),
+            ],
+        );
+        run_test(
+            "Get parsec consensus offline then local event back online",
+            &initial_state,
+            &[LocalEvent::NodeDetectedBackOnline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_votes: vec![ParsecVote::BackOnline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_parsec_offline_and_local_event_back_online_then_parsec_oneline() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[
+                LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event(),
+                ParsecVote::Offline(NODE_ELDER_130).to_event(),
+                LocalEvent::NodeDetectedBackOnline(NODE_ELDER_130).to_event(),
```

Comment:
The local events are not needed here.
Both because they only enqueue votes, but also because that specific node may not have seen that event at all, but it will still process it correctly. (Same for the other tests).

Reply:
Can make the test name shorter as a result.

Reply:
Good point!